### DISCUSSION TITLE
Help with Shelly1 + PIR + Rules

### DISCUSSION THREAD

**Discussion by QBJack** [arendst/Tasmota#19960] [Show and tell]
Hello guys,

I have a Shelly1 (non plus) with a MH-SR602 PIR sensor connected and latest tasmota.

What I try to setup for a few days now but did not succeed is the follwing:

This Shelly + PIR sensor should turn on my stair lights on montion detection and turn them off after 30 seconds. This is working so far as it should.

BUT I would like to have the option to turn the lights on manually AND permanently - they should then not be turned off by the rule that is executed by montion detection from the PIR sensor, which would turn the lights off after 30 seconds.

Please help on that.

Thanks and best regards
Thomas

**Comment by Noschvie** [comment]
Have you had a look to [auto-off-motion-sense-switch](https://tasmota.github.io/docs/Rules/#auto-off-motion-sense-switch) ?

**Reply by QBJack** [reply]
Yes, sure. But it does not work for me as described. I am not sure which switchmode or other settings should be set, as with the suggested Switchmode1 1 the relay is not triggered.

**Comment by QBJack** [comment]
My current rule is this:

ON switch1#state=1 DO Backlog Power1 ON; RuleTimer1 30 ENDON ON Rules#Timer=1 DO Power1 OFF ENDON

and Switchmode1 set to 14.

**Reply by sfromis** [reply]
Read https://tasmota.github.io/docs/PIR-Motion-Sensors/

You can set up additional rules if you want to amend how on/off actions work.

**Reply by QBJack** [reply]
Thanks, but I have already read all tasmota docs about that topic and searched the internet. I have tested so many rule options and settings but nothing worked as desired.

**Reply by sfromis** [reply]
Well, the docs suggestion works fine. No rules.

> Another configuration option is to change Switchmode to 14 with Pulsetime of 130 (30 seconds on every time the AM312 is triggered)

You may have been trying too many different things without returning to start config each time.

**Reply by QBJack** [reply]
This is not what I want to achieve. The pulsetime will also be triggered when I manually turn power1 to on. But it should stay on permanently until switched off manually.

So this is why it needs to be setup with rules IMO.

**Comment by stickilino** [comment]
I solved this with two rules:

SwitchMode1 13
Pulsetime 0

rule1 ON Switch1#state=1 DO Backlog event checksunrise=%time%; event checksunset=%time% ENDON   ON event#checksunrise<%sunrise% DO Backlog Power1 ON; RuleTimer1 5 ENDON   ON event#checksunset>%sunset% DO Backlog Power1 ON; RuleTimer1 5 ENDON ON Rules#Timer=1 DO Power1 OFF ENDON

rule2 on switch1#state=1 do publish PIR ON endon on switch1#state=0 do publish PIR OFF endon

and I changed the active rule via mqtt cmd

**Reply by QBJack** [reply]
Thanks for your help and your rule set.

Unfortunately this does not work so well for me. I have a delay of about 3 to 5 seconds bevor the lights turn on. Maybe the check for sunrise and sunset takes this long to complete? Does this work "immediately" on your setup?

And please tell me why you did mention "I changed the active rule via mqtt cmd"? I do not use any mqtt broker or commands in my setup for now. I set this in the console for each tasmota device manually.

Thanks
Thomas

**Comment by QBJack** [comment]
The console output with your rule (timer set to 30 seconds):

17:54:12.312 RUL: SWITCH1#STATE=1 performs "Backlog event checksunrise=1074; event checksunset=1074"
17:54:12.328 RSL: RESULT = {"Switch1":{"Action":"ON"}}
**17:54:12.347 RSL: RESULT = {"Event":"Done"}
17:54:12.588 RSL: RESULT = {"Event":"Done"}**
17:54:12.626 RUL: EVENT#CHECKSUNSET>%SUNSET% performs "Backlog Power1 ON; RuleTimer1 30"
17:54:12.638 RSL: RESULT = {"POWER":"ON"}
17:54:12.641 RSL: POWER = ON
17:54:12.891 RSL: RESULT = {"T1":30,"T2":0,"T3":0,"T4":0,"T5":0,"T6":0,"T7":0,"T8":0}
17:54:43.409 RUL: RULES#TIMER=1 performs "Power1 OFF"

The bold lines are where the delay occurs I think, but why?

**Reply by QBJack** [reply]
I think the problem is the PIR sensor. With 'switchmode1 1' there is no delay. From the tasmota wiki:

"MH-SR602[~](https://tasmota.github.io/docs/PIR-Motion-Sensors/#mh-sr602)
This is a very small version of a PIR that is able to modify the sensitivity and delay by soldering resistors.

With factory settings this PIR goes to off state after a few seconds (2.5 seconds) so we need to use this rule instead of the one in the example."

But I still have the problem in my rule, that a PIR detection triggers the rule every time, even if the Shelly is turned on manually. And then after 30 seconds, as configured in my rule, the lights turn off. The rule should only run a timer on PIR detection when the relay (light) is OFF.

I tried to achieve this with a BREAK in my rule but this does not work either:

Rule1
ON Power1#state=1 do backlog Power1 BREAK (this rule should stop further rule execution when power=on, but it does not)
ON Switch1#state=1 DO Backlog Power1 ON; RuleTimer1 30 ENDON
ON Rules#Timer=1 DO Power1 OFF ENDON

**Comment by stickilino** [comment]
Hello,
please try the example here:
https://tasmota.github.io/docs/Rules/#enable-a-pir-switch-only-at-night

This discussion was about how to change the lights to be permanently ON
I did this with two rules defined. I control the light via mqtt and openhab. Sending Power and Rule commands to the device.
Power ON permanent is rule2 = ON rule1 = OFF
PIR modus is rule1 =ON and rule2 = OFF

My Rule1 does only work with SwitchMode1 13  ... the example on tasmota.github is SwitchMode1 14 !!!

Please try both
Good Luck
